pandas/core/groupby/generic.py

```diff
@@ -318,13 +318,15 @@ def _aggregate_multiple_funcs(self, arg):
                 obj = copy.copy(obj)
                 obj._reset_cache()
                 obj._selection = name
-            results[name] = obj.aggregate(func)
+            results[base.OutputKey(label=name, position=idx)] = obj.aggregate(func)
 
         if any(isinstance(x, DataFrame) for x in results.values()):
             # let higher level handle
-            return results
+            return {key.label: value for key, value in results.items()}
 
-        return DataFrame(results, columns=columns)
+        if results:
```

Comment:
Is _wrap_aggregated_output failing in results is empty? If so I think that should be fixed rather than branching here